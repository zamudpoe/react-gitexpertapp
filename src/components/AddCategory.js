import React, { useState } from 'react'
import PropTypes from "prop-types"

export const AddCategory = ( { setCategories } ) => {
  const [ inputValue, setInputValue ] = useState("")

  const handleInputChange             = ( e ) => { 
    setInputValue(e.target.value) 

    console.log("HandleInputChange llamado");

  }

  const handleSubmit                  = ( e ) => {  
    e.preventDefault() 
    console.log("%chandleSubmit de <AddCategory /> %s","color: coral; font-size: 1.2rem;", inputValue)

    if ( inputValue.trim().length > 2 ) {  
      setCategories( categories => [ inputValue, ...categories ] )
      /* setCategories( 12345 ) */ // forzamos que falle la prueba con esta linea y comentando la de arriba
      setInputValue("")
    } else {  
      console.log("%cPor favor dame mas de dos letras!","color: tomato;")  
    }  
  }

  return (
    <form onSubmit= { handleSubmit }>
      <h1>{ inputValue }</h1>
      <input 
        type        = "text" 
        value       = { inputValue }
        onChange    = { handleInputChange }        
        onSubmit    = { handleSubmit }
        placeholder = "Introduce un valor"
      />
    </form>
  ) 
}


AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

