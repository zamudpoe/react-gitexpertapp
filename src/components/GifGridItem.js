import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( { id, title, url } ) => {

  /* console.log( { id, title, img } ) */
  /* https://animate.style/ */
  return (
    <div className="card animate__animated animate__zoomInDown" key={ id }>
      <img src={ url } alt={ title } />
      <p> { title } </p>
    </div>
  )

}


/* 
  1. Enzyme
    npm i --save-dev enzyme enzyme-adapter-react-16
  2. Enzyme to JSON
    
  3. PRUEBAS
    *.- Debe moistrar el componente correctamente!.
      *.- Shallow 
      *.- Wrapper
      *- wrapper .toMatchSnapshot()

*/

GifGridItem.propTypes = {
  title : PropTypes.string.isRequired, 
  url   : PropTypes.string.isRequired 
}

