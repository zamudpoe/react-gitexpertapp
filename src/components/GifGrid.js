/* import React, { useState, useEffect, Fragment } from 'react' */
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'

export const GifGrid = ( { category } ) => {  
  const { data:images, loading } = useFetchGifs(category)
  /* console.log(data, loading) */

  return ( 

    <>
      <h3>{ category }</h3> <hr />      

      { loading && <p>Loading...</p> }

      {/* 04 destructuramos el estado y lo enviamos al nuevo componente GifGridItem */}
      <div className="card-grid"> 

        {
          images.map( img => ( 
            <GifGridItem 
              key   = { img.id }  
              { ...img }
            />
          ))
        }

      </div>  
    </>

  ) 
}  


