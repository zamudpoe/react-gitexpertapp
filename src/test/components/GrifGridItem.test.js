import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

const title   = "Ola K Ase?"
const url     = "localhost:3000"
const wrapper = shallow( <GifGridItem title = {title} url = {url} /> )

describe('Pruebas en componente <GrifGridItem />', () => { 

  // Prueba 01
  test('Debe de mostrar <GrifGridItem /> correctamente renderizado!', ( ) => {
    expect( wrapper ).toMatchSnapshot()
  })

  // Prueba 02
  test(`Debe tener un parrafo con el title con valor "${title}"`, () => {
    const p = wrapper.find('p')
    expect( p.text().trim() ).toBe( title )
  })

  // Prueba 03
  test(`Debe tener la imagen igual al url y alt de los props`, () => {

    const img = wrapper.find('img')
    /* console.log( img.prop("alt") ) */
    /* console.log( img.props() ) */

    expect( img.prop("src") ).toBe( url )
    expect( img.prop("alt") ).toBe( title )

  })

  // Prueba 01
  test('Debe tener la clase "animate__zoomInDown" ', () => {

    const div       = wrapper.find('div')
    const className = div.prop("className")

    expect( className.includes('animate__zoomInDown') ).toBe( true ) // una forma con incluye
    expect( div.hasClass("animate__zoomInDown") ).toBe(true) // otra forma mas directa con hasClass

  })

  

})



