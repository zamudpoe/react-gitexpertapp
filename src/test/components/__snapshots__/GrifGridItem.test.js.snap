// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GrifGridItem /> Debe de mostrar <GrifGridItem /> correctamente renderizado! 1`] = `
<div
  className="card animate__animated animate__zoomInDown"
>
  <img
    alt="Ola K Ase?"
    src="localhost:3000"
  />
  <p>
     
    Ola K Ase?
     
  </p>
</div>
`;
